src/Nri/Ui/QuestionBox/V3.elm

```diff
@@ -298,15 +311,37 @@ viewGuidance config referencingId markdown_ =
                     , Css.position Css.relative
                     ]
                 ]
-                [ viewCharacter character_
-                , viewSpeechBubble config
-                    referencingId
-                    [ Balloon.markdown markdown_
-                    , Balloon.onLeft
-                    , Balloon.alignArrowEnd
-                    , Balloon.css [ Css.minHeight (Css.px 46) ]
-                    ]
-                ]
+                ([ config.textToSpeechView
+                    |> Maybe.map
+                        (div
+                            [ css
+                                [ Css.position Css.relative
+                                , Css.zIndex (Css.int 1)
+                                , Css.left (Css.px -24)
+                                , Css.top (Css.px 8)
+                                , withMedia [ quizEngineMobile ]
+                                    [ Css.left Css.auto
+                                    , Css.top Css.auto
+                                    , Css.float Css.left
+                                    , Css.padding4 Css.zero (Css.px 5) Css.zero Css.zero
+                                    , Css.position Css.static
+                                    ]
+                                ]
+                            ]
+                            << List.singleton
```

Comment:
I don't think that the positioning of the tts view should happen from within questionbox. The ReadAloud code is set up to handle positioning (it already has position relative and zindex -- I think in this case, you would override it with position absolute and use top: 8px and left: -24px). Having positioning logic in noredink-ui seems more brittle to me, since if we, for instance, decide we want the readaloud icons to be much bigger, we would then have to do an entire noredink-ui release cycle before we could make the change.

(Why position absolute? I'm noticing that the position: relative is pushing the guidance too far right. Compare the blue space on the left side of the guidance:

<img width="128" alt="Screen Shot 2023-01-27 at 1 39 25 PM" src="https://user-images.githubusercontent.com/8811312/215192681-e2c0aef9-5ff6-45cb-b238-2072be1052b1.png">
<img width="125" alt="Screen Shot 2023-01-27 at 1 39 29 PM" src="https://user-images.githubusercontent.com/8811312/215192684-4d4ce64d-fcfc-49b2-b164-b55f589f3764.png">

It's totally possible that the blue space should increase a bit, based on the figma files I've last seen, but if so, I would guess it should be balanced with the character on the right.

But the character on the right and its positioning is subject to/likely to change! So again, I think having the position styles elsewhere is a good call.
)

Reply:
Nice, I was wondering if this code should be at NRI UI or not, but I think it makes more sense to be outside NRI UI.